src/dataplane-protocol/src/record.rs

```diff
@@ -41,147 +41,119 @@ pub trait AsyncBuffer {
 pub trait Records {}
 
 #[derive(Default)]
-pub struct DefaultAsyncBuffer(Option<Vec<u8>>);
+pub struct DefaultAsyncBuffer(Vec<u8>);
 
 impl DefaultAsyncBuffer {
-    pub fn new(val: Option<Vec<u8>>) -> Self {
+    pub fn new(val: Vec<u8>) -> Self {
         DefaultAsyncBuffer(val)
     }
 
-    pub fn inner_value(self) -> Option<Vec<u8>> {
+    pub fn inner_value(self) -> Vec<u8> {
         self.0
     }
 
-    pub fn inner_value_ref(&self) -> &Option<Vec<u8>> {
+    pub fn inner_value_ref(&self) -> &Vec<u8> {
```

Comment:
Is this used anywhere? Generally `&Vec<T>` is an anti-pattern. If this function is actually needed, can we get away with turning it into `fn as_bytes(&self) -> &[u8]` instead?

Reply:
I think that's a good thing for the next refactor. This PR has touched way more of the codebase than I wanted it to.